applications/CompressiblePotentialFlowApplication/custom_elements/transonic_perturbation_potential_flow_element.cpp

```diff
@@ -0,0 +1,856 @@
+    }
+    else // Wake element
+    {
+        if (rResult.size() != 2 * NumNodes)
+            rResult.resize(2 * NumNodes, false);
+
+        GetEquationIdVectorWakeElement(rResult);
+    }
+}
+
+template <int Dim, int NumNodes>
+void TransonicPerturbationPotentialFlowElement<Dim, NumNodes>::GetDofList(DofsVectorType& rElementalDofList,
+                                                                 ProcessInfo& CurrentProcessInfo)
+{
+    const TransonicPerturbationPotentialFlowElement& r_this = *this;
+    const int wake = r_this.GetValue(WAKE);
+
+    if (wake == 0) // Normal element
+    {
+        if (rElementalDofList.size() != NumNodes)
+            rElementalDofList.resize(NumNodes);
+
+        const int kutta = r_this.GetValue(KUTTA);
+
+        if (kutta == 0)
+            GetDofListNormalElement(rElementalDofList);
+        else
+            GetDofListKuttaElement(rElementalDofList);
+    }
+    else // wake element
+    {
+        if (rElementalDofList.size() != 2 * NumNodes)
+            rElementalDofList.resize(2 * NumNodes);
+
+        GetDofListWakeElement(rElementalDofList);
+    }
+}
+
+template <int Dim, int NumNodes>
+void TransonicPerturbationPotentialFlowElement<Dim, NumNodes>::FinalizeSolutionStep(ProcessInfo& rCurrentProcessInfo)
+{
+    bool active = true;
+    if ((this)->IsDefined(ACTIVE))
+        active = (this)->Is(ACTIVE);
+
+    const TransonicPerturbationPotentialFlowElement& r_this = *this;
+    const int wake = r_this.GetValue(WAKE);
+
+    if (wake != 0 && active == true)
+    {
+        ComputePotentialJump(rCurrentProcessInfo);
+    }
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+// Inquiry
+
+template <int Dim, int NumNodes>
+int TransonicPerturbationPotentialFlowElement<Dim, NumNodes>::Check(const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY
+
+    // Generic geometry check
+    int out = Element::Check(rCurrentProcessInfo);
+    if (out != 0)
+    {
+        return out;
+    }
+
+    KRATOS_ERROR_IF(GetGeometry().Area() <= 0.0)
+        << this->Id() << "Area cannot be less than or equal to 0" << std::endl;
+
+    for (unsigned int i = 0; i < this->GetGeometry().size(); i++)
+    {
+        KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(VELOCITY_POTENTIAL, this->GetGeometry()[i]);
+    }
+
+    return out;
+
+    KRATOS_CATCH("");
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+
+template <int Dim, int NumNodes>
+void TransonicPerturbationPotentialFlowElement<Dim, NumNodes>::GetValueOnIntegrationPoints(
+    const Variable<double>& rVariable, std::vector<double>& rValues, const ProcessInfo& rCurrentProcessInfo)
+{
+    if (rValues.size() != 1)
+        rValues.resize(1);
+
+    if (rVariable == PRESSURE_COEFFICIENT)
+    {
+        rValues[0] = PotentialFlowUtilities::ComputePerturbationCompressiblePressureCoefficient<Dim, NumNodes>(*this, rCurrentProcessInfo);
```

Comment:
```suggestion
        rValues[0] = PotentialFlowUtilities::ComputePerturbationCompressiblePressureCoefficient<Dim, NumNodes>(rCurrentProcessInfo);
```
You must not pass the `this` as you can access to "yourself" from any part inside the scope of the class.